Runner.java

```diff
@@ -0,0 +1,41 @@
+import beans.ExtraTrial;
+import beans.LightTrial;
+import beans.StrongTrial;
+import beans.Trial;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Runner {
+    public static void main(String[] args) {
+        Trial[] array = new Trial[]{
+                new Trial("Trial1", 35, 60),
+                new Trial("Trial2", 53, 46),
+                new Trial("Trial3", 27, 16),
+                new LightTrial("LightTrial1", 84, 95),
+                new LightTrial("LightTrial2", 76, 59),
+                new StrongTrial("StrongTrial1", 83, 78),
+                new StrongTrial("StrongTrial2", 64, 73),
+                new ExtraTrial("ExtraTrial1", 53, 75, 69),
+                new ExtraTrial("ExtraTrial2", 42, 36, 75)
+        };
+        List<Trial> trials = Arrays.asList(array);
+
+        trials.stream().forEach(System.out::println);
+
+        int passedCounter = (int) trials.stream().filter(Trial::isPassed).count();
+        System.out.println(passedCounter);
+
+        trials.sort(Comparator.comparingInt(Trial::getMarkSum));
+
+        trials.stream().map(Trial::getMarkSum).forEach(System.out::println);
```

Comment:
`getMarkSum` is an unnatural method for `Trial`.
if it were protected i would agree. 
you made it public then naturally to override it in `ExtraTrial`. 
but this does not suit for `Runner`. 
so the solution is to define a local lambda `t -> (t.getMark1() + t.getMark2()` in the `main` method and then use it  instead of `Trial::getMarkSum`.